src/preamble.js

```diff
@@ -577,7 +583,7 @@ function resetPrototype(constructor, attrs) {
 }
 #endif
 
-#if !SOURCE_PHASE_IMPORTS
+#if !SOURCE_PHASE_IMPORTS && !WASM_ESM_INTEGRATION
```

Comment:
Can these two settings not be merged into one? Are there use cases for one without the other?

Reply:
They are fairly different.  I think it make sense to keep them separate.   

I'm not big of something like `ESM=1` and `ESM=2` where the numbers don't really describe which is happening.

Reply:
Why would I was source phase imports but not wasm ESM, though?

Reply:
What about trying to move all the settings that control output format into one? maybe `OUTPUT_FORMAT`?
then have:
 - esm
 - es
 - source-phase
 - cjs
 - none
 - ....

Reply:
> What about trying to move all the settings that control output format into one? maybe `OUTPUT_FORMAT`? then have:
> 
> * esm
> * es

Are you suggesting `es` for the existing JS module output?

The problem is that they are both ESM, right?  So we could need `esm` and `esm-with-wasm-integration` both I think.

Reply:
> Why would I want source phase imports but not wasm ESM, though?

Source phase imports is basically a different type of ESM integration.   Its more flexible and likely to be supported in browsers soon.  Full ESM integration is different, and further from browser adoption, but its available in node today with certain flags. 

The both allow is to remove loading code but full ESM integration allows to remove even more.

For now at least, I think they are separately useful.